OptimoveSDK/Sources/Classes/Optimove.swift

```diff
@@ -46,20 +46,40 @@ typealias Logger = OptimoveCore.Logger
     public static func initialize(with config: OptimoveConfig) {
         shared.config = config
 
-        if config.isOptimoveConfigured(), let tenantInfo = config.tenantInfo {
-            Optimove.configure(for: tenantInfo)
+        if config.isOptimoveConfigured() {
+            Optimove.configure(for: config)
         }
 
-        if config.isOptimobileConfigured(), let optimobileConfig = config.optimobileConfig {
+        if config.isOptimobileConfigured() {
             shared.container.resolve { serviceLocator in
-                guard let visitorId = try? serviceLocator.storage().getInitialVisitorId() else {
-                    return
+                do {
+                    let visitorId = try serviceLocator.storage().getInitialVisitorId()
+                    let userId = try? serviceLocator.storage().getCustomerID()
+
+                    Optimobile.initialize(config: config, initialVisitorId: visitorId, initialUserId: userId)
+                } catch {
+                    throw GuardError.custom("Failed on OptimobileSDK initialization. Reason: \(error.localizedDescription)")
                 }
+            }
+        }
+    }
 
-                let userId = try? serviceLocator.storage().getCustomerID()
+    /// Set the credentials for the Optimove server. Intent to use as a step for the delayed initialization.
+    public static func setCredentials(optimoveCredentials: String?, optimobileCredentials: String?) {
+        let builder = OptimoveConfigBuilder(optimoveCredentials: optimoveCredentials, optimobileCredentials: optimobileCredentials)
+        builder.setFeatures([builder.features, .delayedConfiguration])
```

Comment:
another difference here is that in android we dont update features after initial initialisation. If you initially say you want optimobile only and then give optimove credentials/dont give optimobile -- we throw. That's 1 of goals of FeatureSet -- telling early what do they want so that later can make sure they gave credentials for that

Reply:
This part was changed: added a new OptimoveConfigBuilder init with for the current config -- it saves all existed settings.